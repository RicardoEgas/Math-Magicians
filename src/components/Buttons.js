import React from 'react';
import './styles.css';
import PropTypes from 'prop-types';

const Buttons = ({ number, handleEvent }) => {
  if (number === '\u00F7' || number === 'x' || number === '-' || number === '+' || number === '=') {
    return <div onClick={handleEvent} onKeyDown={handleEvent} role="presentation" className="btn orange">{number}</div>;
  } if (number === '0') {
    return <div onClick={handleEvent} onKeyDown={handleEvent} role="presentation" className="btn zero">{number}</div>;
  }
  return <div onClick={handleEvent} onKeyDown={handleEvent} role="presentation" className="btn">{number}</div>;
};

Buttons.propTypes = {
  number: PropTypes.string.isRequired,
  handleEvent: PropTypes.func.isRequired,
};

export default Buttons;
