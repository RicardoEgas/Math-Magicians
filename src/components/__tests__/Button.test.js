import React from 'react';
import { render } from "@testing-library/react";
import Buttons from '../Buttons'; // Adjust the path

describe('Buttons component', () => {
  it('should render with the provided number and handle event', () => {
    const handleEventMock = jest.fn();

    const { tree } = render(<Buttons number="1" handleEvent={handleEventMock} />);
    expect(tree).toMatchSnapshot();
  });

  // Add more test cases as needed
});
