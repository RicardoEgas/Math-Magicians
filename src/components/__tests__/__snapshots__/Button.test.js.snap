// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component should render with the provided number and handle event 1`] = `undefined`;
